### PR TITLE
 PS-5216: Fix utility user invisibility issues, and document limitations (5.7)

### PR DESCRIPTION
**Pull request by dutow**

### CODE REVIEW

File: storage/perfschema/unittest/pfs_server_stubs.cc [hunk 1]
```diff
@@ -44,3 +44,9 @@ void sql_print_warning(const char *format, ...)
 {
   /* Do not pollute the unit test output with annoying messages. */
 }
+
+my_bool
+acl_is_utility_user(const char *user, const char *host, const char *ip)
```
Comment (percona-ysorokin) [inline storage/perfschema/unittest/pfs_server_stubs.cc]: Please, double check this does not trigger any `unused arguments` warnings. This concerns 5.6 as well

**Review by percona-ysorokin — APPROVED** [review]
LGTM

**Review by percona-ysorokin — APPROVED** [review]
LGTM

File: sql/auth/sql_auth_cache.cc [hunk 2]
```diff
@@ -1606,11 +1606,13 @@ acl_init_utility_user(bool check_no_resolve)
     {
       if (*cur_pos == ',' || *cur_pos == '\0')
       {
-        acl_utility_user_schema_access.push_back(std::string(cur_db,
-                                                             cur_pos
-                                                             - cur_db));
+	if (cur_pos - cur_db > 1) {
```
Comment (percona-ysorokin) [inline sql/auth/sql_auth_cache.cc]: Shouldn't there be `cur_pos - cur_db >= 1`?

Comment (dutow) [reply sql/auth/sql_auth_cache.cc]: No, that still would include the empty entries - if the length of this string is one, it is the comma, or the end of string character.

Comment (percona-ysorokin) [reply sql/auth/sql_auth_cache.cc]: consider the input `a,b`
when `cur_pos` points to the comma `,`,  `cur_db` points to the beginning of the string `a`
in this case `cur_pos - cur_db == 1`, we won't get into this branch

Comment (dutow) [reply sql/auth/sql_auth_cache.cc]: Sorry, you are right. And I did this correctly in 5.6, it's only incorrect on 5.7...

**Comment by dutow** [comment]
https://ps57.cd.percona.com/job/percona-server-5.7-param/150/

**Pull request merged**